### ISSUE TITLE
[mesheryctl] Helm: cannot start Meshery: rendered manifests contain a resource that already exists.

### ISSUE THREAD

**Issue by leecalcote** [meshery/meshery#4777]
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior
As a contributor, I have built Meshery Server locally. Meshery Server connects to my Kubernetes cluster and deploys Meshery Operator using helm.

When I cancel the locally-running Meshery Server process, the Meshery Operator deployment stays residual on my Kubernetes cluster.

Then, when I use `mesheryctl system start` to start Meshery as a user would, I receive the error shown in the logs below.

#### Desired Behavior
`mesheryctl system start` should have idempotent behavior in terms of installing Meshery Operator upon connecting to a Kubernetes cluster.

#### Screenshots/Logs
```
$ mesheryctl system start
Starting Meshery...
⣯ Deploying Meshery on Kubernetes Error: failed to start Meshery
See https://docs.meshery.io/reference/mesheryctl/system for usage details
: cannot start Meshery: rendered manifests contain a resource that already exists. Unable to continue with install: ServiceAccount "meshery-operator" in namespace "meshery" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: key "meta.helm.sh/release-name" must equal "meshery": current value is "meshery-operator"
```

#### Environment

- Host OS: Mac
- Meshery Server Version: edge
- Meshery Client Version: stable-v0.6.0-rc-2
- Platform: Kubernetes

---

#### Contributor Resources
- [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

**Comment by leecalcote** [comment]
@panyuenlau I wonder if you want to bite into this one.

If not, @hexxdump or @piyushsingariya is this something you can address?

**Comment by panyuenlau** [comment]
@leecalcote quick question, how did you install and stop the meshery server before running `system start`? If the meshery-operator was installed by the meshery server in the first place, shouldn't the meshery server also remove the meshery-operator deployment when we stop the meshery server?

**Comment by leecalcote** [comment]
@panyuenlau, yes, you're right. Here's a quick recap from the sequence of events in the issue description:

1. Contributor locally compiles Meshery Server; runs as a process on local machine.
2. Meshery Server connects to K8s; deploys Meshery Operator.
3. Contributor kills Meshery Server process mid-flight; non-graceful shutdown of Meshery Server.
4. Meshery Operator deployment stays residual on contributor's Kubernetes cluster.
5. User executes `mesheryctl system start` against the same K8s cluster; receives error.

While the sequence described above is described as either 1) only affecting contributors or 2) only affecting a Kubernetes cluster that both a contributor and a user share, the more critical user flows to support are: 

1) idempotency of a user running `mesheryctl system start` against the same cluster more than once.
2) idempotency of one or more cluster admins running the `system start` command against the same cluster (unbeknownst to one another - not as a race condition, but in that each user assumed Meshery (Operator) wasn't already present).
3) in context of multiple K8s cluster support in which `mesheryctl` or Meshery Server might need to re-connect to previously connected cluster(s) and will need to gracefully handle the situation in which an Operator is already present.
- yet to be implemented: [Design Spec: Supporting multiple Kubernetes clusters in mesheryctl](https://docs.google.com/document/d/1r_Yopt904qdqO6lutzZn8mfqjKlI-MQgD22GG8x4UYM/edit) // @navendu-pottekkat 
- implemented and currently under review: Multi-context (cluster) support in Meshery Server - https://github.com/meshery/meshery/pull/4218 // @utkarsh-pro

**Comment by piyushsingariya** [comment]
If this issue is mainly faced by community developers. maybe we can implement a process which deletes Meshery-operator resources after someone stops locally compiled Meshery. We could trigger the same API/handler that's used to deploy operator through UI just to remove the resources.

Does this make sense?? I feel this could be implemented in similar fashion where we delete the cached patters. filters after stopping Meshery.

**Comment by navendu-pottekkat** [comment]
Related issue: https://github.com/meshery/meshery/issues/4773
